### PR TITLE
swkbd: Fix a bug where clicking Cancel hangs the game

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
The text is validated in `Finalize`. If the validation fails, an error is returned and the applet is not actually finalized. This can result in hangs.

This is usually not a problem as the frontend is expected to validate the text passed to `Finalize`. However, when the user clicked on `Cancel`, the text is ignored and the frontend won't do any validation. Therefore, we should skip the validation here as well.

Also fixed a potential data race. All these functions should now be called on the same thread

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5294)

### CODE REVIEW

File: src/citra_qt/applets/swkbd.h [hunk 1]
```diff
@@ -62,6 +62,9 @@ class QtKeyboard final : public QObject, public Frontend::SoftwareKeyboard {
 
     QWidget& parent;
 
+    std::string text;
```
Comment (B3n30) [inline src/citra_qt/applets/swkbd.h:65]: NIT: those member names are very generic and could probably be a bit more expressive

**Comment by Dragios** [comment]
The moment I close the dialog via `X` button it will suddenly remove any input in the Nickname column as shown in red box even when there is words already previously saved.

**Comment by Dragios** [comment]
I am wondering if the `X` button should even be exist at all for HLE keyboard?
The first time you boot up the game Luxor, you are required to create a new profile, hence no cancel button in HLE keyboard. On real 3DS, you cannot even exit to the Home Menu, if the swkbrd is still active. But on HLE if you `X` out the dialog, it will reprompt again until you input something and hit OK (which is also fine I guess).

If we were to imitate as close to the actual 3DS behavior, the Close dialog button should be muted or not exist at all. Of course, this can be up for debate.
What are other developer's opinion on this?

**Comment by zhaowenlan1779** [comment]
@Dragios Fixed. now the close button should be disabled.

**Pull request merged**